provider/identity_provider_resource.go

```diff
@@ -336,6 +453,25 @@ func (r *IdentityProviderResource) Create(ctx context.Context,
 		}
 		gitlabBuilder.URL(state.Gitlab.URL.Value)
 		builder.Gitlab(gitlabBuilder)
+	case state.Github != nil:
+		builder.Type(cmv1.IdentityProviderTypeGithub)
+		githubBuilder := cmv1.NewGithubIdentityProvider()
+		ghState := state.Github
+		githubBuilder.ClientID(ghState.ClientID.Value)
+		githubBuilder.ClientSecret(ghState.ClientSecret.Value)
+		if !ghState.CA.Unknown && !ghState.CA.Null {
+			githubBuilder.CA(ghState.CA.Value)
+		}
+		if !ghState.Hostname.Unknown && !ghState.Hostname.Null && len(ghState.Hostname.Value) > 0 {
+			githubBuilder.Hostname(ghState.Hostname.Value)
+		}
+		if len(ghState.Teams) > 0 {
```

Comment:
teams is Optional: true,
you should check  if it's not unknown and not nil

Reply:
Teams, and orgs are string arrays. Not a tf typed struct. It has no `.Unknown` nor `.Null`, also since it's not a pointer, the default is an empty array. [ref](https://github.com/terraform-redhat/terraform-provider-ocm/pull/53/files#diff-6d537d29d94410c13a4ae530df04cbf7ab877d0a3c1ff94a5fe86850a96e2bb6R51). In this case, Optional is only a validation. 

Tests can validate this as well.

Reply:
Please add a subsystem test that covers it.